src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php

```diff
@@ -12,31 +12,89 @@
 namespace Symfony\Bundle\WebProfilerBundle\Twig;
 
 use Symfony\Component\HttpKernel\DataCollector\Util\ValueExporter;
+use Symfony\Component\VarDumper\Cloner\Data;
+use Symfony\Component\VarDumper\Dumper\HtmlDumper;
 
 /**
  * Twig extension for the profiler.
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
-class WebProfilerExtension extends \Twig_Extension
+class WebProfilerExtension extends \Twig_Extension_Profiler
 {
     /**
      * @var ValueExporter
      */
     private $valueExporter;
 
     /**
+     * @var HtmlDumper
+     */
+    private $dumper;
+
+    /**
+     * @var resource
+     */
+    private $output;
+
+    /**
+     * @var int
+     */
+    private $stackLevel = 0;
+
+    public function __construct(HtmlDumper $dumper = null)
+    {
+        $this->dumper = $dumper ?: new HtmlDumper(null, null, HtmlDumper::DUMP_LIGHT_ARRAY);
+        $this->dumper->setOutput($this->output = fopen('php://memory', 'r+b'));
+    }
+
+    public function enter(\Twig_Profiler_Profile $profile)
+    {
+        ++$this->stackLevel;
+    }
+
+    public function leave(\Twig_Profiler_Profile $profile)
```

Comment:
The use case is changing the output stream of the HtmlDumper so that it resets its `headerIsDumped` internal state and can be reused after use.